packages/api/src/moderation/accumulator.ts

```diff
@@ -0,0 +1,181 @@
+import { AppBskyGraphDefs } from '../client/index'
+import {
+  Label,
+  LabelPreference,
+  ModerationCause,
+  ModerationOpts,
+  ModerationDecision,
+} from './types'
+import { LABELS } from './const/labels'
+
+export class ModerationCauseAccumulator {
+  did = ''
+  causes: ModerationCause[] = []
+
+  constructor() {}
+
+  setDid(did: string) {
+    this.did = did
+  }
+
+  addBlocking(blocking: string | undefined) {
+    if (blocking) {
+      this.causes.push({
+        type: 'blocking',
+        source: { type: 'user' },
+        priority: 3,
+      })
+    }
+  }
+
+  addBlockedBy(blockedBy: boolean | undefined) {
+    if (blockedBy) {
+      this.causes.push({
+        type: 'blocked-by',
+        source: { type: 'user' },
+        priority: 4,
+      })
+    }
+  }
+
+  addLabel(label: Label, opts: ModerationOpts) {
+    // look up the label definition
+    const labelDef = LABELS[label.val]
+    if (!labelDef) {
+      // ignore labels we don't understand
+      return
+    }
+
+    // look up the label preference
+    // TODO use the find() when 3P labelers support lands
```

Comment:
I came here to say "oh wait will it be a heavy lift to add 3P labeler support" and the API signature is already in place :kissing_heart: